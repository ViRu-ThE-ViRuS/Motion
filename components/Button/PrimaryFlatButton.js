import React, { Component } from 'react';
import {
    View,
    Text,
    TouchableOpacity,
    StyleSheet,
} from 'react-native';

export default class PrimaryFlatButton extends Component {
    render() {
        return (
          <TouchableOpacity style={styles.container} onPress={this.props.onPress}>
            <View>
              <Text style={styles.text}>{this.props.text}</Text>
            </View>
          </TouchableOpacity>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        height: 50,
        alignSelf: 'stretch',
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: '#5C7D87',
    },
    text: {
        fontFamily: 'SourceSansPro_Regular',
        color: '#fff',
        fontSize: 15,
    },
});
