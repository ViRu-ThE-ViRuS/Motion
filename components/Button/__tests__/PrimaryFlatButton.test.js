import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import PrimaryFlatButton from './../PrimaryFlatButton';

describe('PrimaryFlatButton tests', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <PrimaryFlatButton text="Get Started" />
    );

    expect(tree).toMatchSnapshot();
  });
});
