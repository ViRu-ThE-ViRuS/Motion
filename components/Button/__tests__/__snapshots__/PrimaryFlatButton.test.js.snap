// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryFlatButton tests renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#5C7D87",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "SourceSansPro_Regular",
          "fontSize": 15,
        }
      }
    >
      Get Started
    </Text>
  </View>
</View>
`;
